### PR TITLE
Make bigtable::Table CopyAssignable.

### LINKED ISSUE #408: Make bigtable::Table CopyAssignable and CopyConstructible
There is no reason why this class should not meet the [CopyAssignable](http://en.cppreference.com/w/cpp/concept/CopyAssignable) and [CopyConstructible](http://en.cppreference.com/w/cpp/concept/CopyConstructible) requirements.

It is a matter of storing the policies as shared pointers instead of unique pointers.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #408.

### CODE REVIEW

File: bigtable/client/table_test.cc [hunk 1]
```diff
@@ -49,3 +49,33 @@ TEST_F(TableTest, TableConstructor) {
...
+  bigtable::Table source(client_, "my-table");
+  std::string expected = source.table_name();
+  bigtable::Table copy(std::move(source));
+  EXPECT_EQ(expected, copy.table_name());
+}
+
+TEST_F(TableTest, CopyAssignment) {
+  bigtable::Table source(client_, "my-table");
+  std::string expected = source.table_name();
+  bigtable::Table dest(client_, "anpother-table");
```
Comment (paprat) [inline bigtable/client/table_test.cc]: "another-table"

File: bigtable/client/internal/table_test.cc [hunk 2]
```diff
@@ -87,6 +87,36 @@ TEST_F(TableTest, TableConstructor) {
...
+  bigtable::noex::Table source(client_, "my-table");
+  std::string expected = source.table_name();
+  bigtable::noex::Table copy(std::move(source));
+  EXPECT_EQ(expected, copy.table_name());
+}
+
+TEST_F(TableTest, CopyAssignment) {
+  bigtable::noex::Table source(client_, "my-table");
+  std::string expected = source.table_name();
+  bigtable::noex::Table dest(client_, "anpother-table");
```
Comment (paprat) [inline bigtable/client/internal/table_test.cc]: "another-table"

File: bigtable/client/table_test.cc (hunk 1 above)
Comment (coryan) [reply bigtable/client/table_test.cc]: Fixed, thanks!

File: bigtable/client/internal/table_test.cc (hunk 2 above)
Comment (coryan) [reply bigtable/client/internal/table_test.cc]: Fixed.

**Pull request merged**